enhancements/installer/operators-check.md

```diff
@@ -0,0 +1,174 @@
+seems low and a risk that could be managed by monitoring these specific failures
+in CI.
+
+### Drawbacks
+
+This enhancement has some significant potential drawbacks in terms of design:
+
+A fundamental design principle of 4.x and the operator framework has been
+delegating responsibility away from the Installer to allow clean separation
+of concerns and better maintainbility. Without this enhancement, it is the
+responsibility of the Cluster-Version Operator to determine whether given
+Cluster Operators constitute a successful version. The idea of keeping the
+cluster-version operator as the single responsible party is discussed in the
+alternatives section.
+
+Following the previous point, this enhancement could introduce a slippery slope.
+Where before this enhancement, the Installer would simply query the CVO; now we
+are adding additional conditional logic. Introducing this conditional logic leads
+to the risk of further conditions, which could make the install process more brittle
+and introduce the potential for false positives or other failures.
+
+Does implementing this enhancement address symptoms of issues with operator status definitions?
+Should operators themselves be setting Degraded=True when they don't meet this stability criterion?
+
+As we have seen with other timeouts in the Installer, developers and users will want to change these.
+We should define a process for refining our stability criteria. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. What is the correct definition of a stable operator? More importantly, how can we
+refine this definition?
+2. Should this logic belong in the Installer, CVO, or another component?
+
+### Test Plan
+
+This code would go directly into the installer and be exercised by e2e tests.
+
+### Graduation Criteria
+
+
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+The cluster-version operator can institute similar logic and trigger alerts when
+operator stability is in question during an upgrade. See alternatives section.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+- The worst case failure more is that the Installer throws an error when there is not an actual
+problem with a cluster. In this case, an admin would need to investigate an error or automation would
+need to rerun an install. We would hope to eliminate this failures through monitoring CI.
+
+#### Support Procedures
+
+We should instruct users on how to debug or review their operators when this error occurs.
+
+## Implementation History
+
+N/A 
+
+## Alternatives
+
+We should seriously consider whether this logic should go into the cluster-version
+operator rather than the installer. (Or if it should be in both, perhaps just the CVO
+for upgrades.) Management of Cluster Operators has been the responsibility of the CVO
+and the Installer should defer responsibility where possible to maintain separation
+of concerns.
```

Comment:
> What is the argument against putting this in the CVO, which lead to this EP?

I'm not sure if there is an argument against putting it in the CVO. I read through the arch call notes, & I couldn't find or remember a strong argument against it. If CVO folks don't have an objection, this enhancement should be more upfront about the openness of this question. I would like this enhancement to serve to make that decision.

Reply:
I could see the argument for "CVO doesn't know/care that we're doing an install, specifically", so that might be one argument for putting the "ok time to watch for 30s of stability within a 5 min timeout" logic in the installer, but yeah, sounds like we should at least sketch out a little deeper what it would mean for the CVO to handle this.

Reply:
> Is the idea behind moving it to the CVO to let the installer only watch the ClusterVersion resource?

if this is directed at me:  yes, that was my motivation in probing on that approach more, though as i later noted, there is also a downside that now the CVO needs to be aware that we're in "install mode" so that it can verify we have a period of stability (or maybe the CVO always reports an aggregate progressing stability condition.....  progressing_stable=true whenever no CO has been progressing=true in the last 30s.)

Reply:
I think at this point there's decent consensus that, for now, we'll just have Installer scan all ClusterOperator conditions. I'm not sure how much more we need to dig into this alternative given we're not immediately proposing that the CVO take this up.